### ISSUE TITLE
connection limit (possible duplicate of #151)

### ISSUE THREAD

**Issue by camypaj** [gpodder/gpodder#340]
Hi all,

I've noticed an unusual bug in v3.9.0 (2016-02-03) (installed on Linux Mint Cinnamon 18.2).
Downloads are by default limited to 1, and that works. But if I increase the number (with checkbox enabled), nothing changes (it still downloads only one file). Only when I unticked the checkbox, it starts downloading in parallel. However, if I check the box again, it continues to download them all.
I just double-checked, with another podcast, and it still happens.
Can anyone test/confirm with the latest version?

Thanks!

**Comment by elelay** [comment]
Thanks for the report.
In current gtk3 branch, this is inverted:
 - increasing the limit will download 1 more file in parallel, as expected
 - unchecking **Limit downloads** doesn't change anything.

**Comment by elelay** [comment]
Checked on master branch f45ee075 :
Once episode downloads are started, they are never cancelled due to download count limit decrease or activation

with this in mind,
 1. increasing the limit by 1 will start downloading 1 queued file (as expected)
 2. checking the "limit downloads" (with limit < current download count) checkbox will let now excessive downloads finish but won't start new ones (as expected)
 3. decreasing the limit will let now excessive downloads finish but won't start new ones (as expected)
 - increasing the limit by 10 will only start downloading 1 queued file (**broken**)
 4. unchecking the "limit downloads" checkbox won't start downloading pending downloads (**broken**)

A quick fix can make 4. do like 3. : start 1 queued download, but it's not enough.

Current download limit needs to be refactored to:
 - allow starting as many downloads as allowed when increasing or disabling limit
 - handle downloads, sync, send to local folders separately (see closed PR #300)

**Comment by elelay** [comment]
> maybe also address #151 to put a hard limit on parallel downloads

fixed in PR #448
